react-context/flow.txt

```diff
@@ -0,0 +1,136 @@
+window.location to the new URL we made, with the updated hash. This is
+effectively changing what we view on the screen to the #sign-in page.
+
+I don’t understand why we are back in app.jsx.
+Maybe we’re back here because the state of route changed?
+But now we are back in app.jsx and re-rendering the page, which is calling
+the this.renderPage method. But now, this.state.path is not equal to an
+empty string. It is now equal to “sing-up”. So, renderPage returns
+the <Auth /> element, from the auth.jsx file.
+
+In auth.jsx, we are pulling the data out of AppContext. Since the user is still
+null, we move on to generating the react elements of the page. Based on
+route.path being === “sign-in”, we are saying “Please Sign in”. If the path
+were “sign-up”,  we’d be saying “Make an account”. So, we start creating an
+element with the appropriate message for the user. Part of the process of
+making the element calls for the react element <AuthForm /> with the properties
+key, action, and onSignIn.
+
+In the auth-form.jsx we are importing the props passed from auth.jsx and giving
+the state object of the elements an empty username and an empty password. The
+render method of the class then generates the form needed, with the appropriate
+labels for signing in or creating an account. This form is being created with
+event listeners for change in the boxes and a button for submit. The created
+form gets returned to the auth.jsx file which then returns the entire sign-in
+page to renderPage method of the App class. Then the element gets inserted into
+the return of the render of App. We now have a sign in page on the screen.
+
+The user clicks Sign-up, causing the this.state.route to change, triggering the
+componentDidMount() of the Apps class to run again. The Apps class goes through
+the same steps as before, but this time, the final elements being rendered are
+labeled oppositely, for signing up. The user inputs their username and password,
+triggering the onChange events of the form, causing the value of the form input
+elements to update as the user types.
+
+There is a prop being passed to the form element called action. This prop’s
+value is based on the pathway the user is currently on.
+
+User can sign up:
+When the user hits “Register”, a fetch request is called with the POST method,
+to the URL “/api/auth/${action}”, in this case, the url would be
+“/api/auth/sign-up”. This goes over to the server, to index.js. There, we are
+using express to manage the request.
+1) First we check if the request’s body property has a username and a password.
+2) Then, we hash the password with the argon2 object.
+3) When the has method is done, we insert into the database the username and
+the hashed passwords, using params, of course.
+4) This same SQL command returns the userId, username, and the time of creation
+5) When that sql query is done, we send a 201 status code to the user and
+provide them with the info we returned from SQL.
```

Comment:
Yes, but be careful not to confuse the terms "user" and "client". The client is the program that sent the request and received the response, whereas the user is the human using the client program.